doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+| `saas`   | `<SAAS_ACCOUNT_ID>`  | yes       | -              | Account ID for accessing an SaaS database instance                 |
+| `saas`   | `<SAAS_DATABASE_ID>` | yes       | -              | Database ID of an Exasol SaaS database instance                    |
+| `saas`   | `<SAAS_PAT>`         | yes       | -              | Personal access token for accessing an SaaS database instance      |
+
+### AAF Python Package
+
+The latest version of AAF can be obtained from [pypi](https://pypi.org), see also the [releases on GitHub](https://github.com/exasol/advanced-analytics-framework/releases).
+
+The following command installs the AAF from pypi:
+
+```bash
+pip install exasol-advanced-analytics-framework
+```
+
+### Script Language Containers (SLCs)
+
+Exasol executes User Defined Functions (UDFs) in an isolated Script Language Container (SLCs).
+Running the AAF requires a custom SLC.
+
+#### Build the SLC
+
+If you want to build the AAF SLC then
+
+1. Download the [AAF Sources](https://github.com/exasol/advanced-analytics-framework/) from GitHub
+2. Install poetry
+3. Use the AAF developer commands for building and deploying custom SLCs into Exasol.
+
+```shell
+poetry run nox build_language_container
+```
+
+See [Install the SLC](#install-the-slc).
+
+#### Download a Pre-built SLC
+
+As an alternative to building SLC yourself you can also download a prebuilt AAF SLC from the [AAF releases](https://github.com/exasol/advanced-analytics-framework/releases/latest) on GitHub.
+
+#### Install the SLC
+
+Installing the SLC requires loading the container file into the BucketFS and registering it to the database:
+
+```shell
+SLC_FILE=.slc/exasol_advanced_analytics_framework_container_release.tar.gz
+LANGUAGE_ALIAS=PYTHON3_AAF
+python -m exasol_advanced_analytics_framework.deploy language-container \
+    --dsn <DB_HOST:DB_PORT> \
+    --db-user <DB_USER> \
+    --db-pass <DB_PASSWORD> \
+    --bucketfs-name <BUCKETFS_NAME> \
+    --bucketfs-host <BUCKETFS_HOST> \
+    --bucketfs-port <BUCKETFS_PORT> \
+    --bucketfs-user <BUCKETFS_USER> \
+    --bucketfs-password <BUCKETFS_PASSWORD> \
+    --bucket <BUCKETFS_NAME> \
+    --path-in-bucket <PATH_IN_BUCKET> \
+    --language-alias "$LANGUAGE_ALIAS" \
+    --container-file "$SLC_FILE"
+```
+
+### Additional Scripts
+
+Besides the BucketFS connection, the SLC, and the Python package AAF also requires some additional Lua scripts to be created in the Exasol database.
+
+The following command deploys the additional scripts to the specified `DB_SCHEMA` using the `LANGUAGE_ALIAS` of the SLC:
+
+```shell
+python -m exasol_advanced_analytics_framework.deploy scripts \
+    --dsn <DB_HOST:DB_PORT> \
+    --db-user <DB_USER> \
+    --db-pass <DB_PASSWORD> \
+    --schema <DB_SCHEMA> \
+    --language-alias "$LANGUAGE_ALIAS"
+```
+
+## Usage
+
+The entry point of this framework is `AAF_RUN_QUERY_HANDLER` script. This script is simply a query loop which is responsible for executing the implemented algorithm.
```

Comment:
We might need to explain that an algorithm is implemented as QueryHandler and what a QueryHandler is.

Reply:
Proposal: Move to separate PR.